src/constraints/absorb.rs

```diff
@@ -8,53 +8,82 @@ use ark_r1cs_std::groups::curves::short_weierstrass::{
     AffineVar as SWAffineVar, ProjectiveVar as SWProjectiveVar,
 };
 use ark_r1cs_std::groups::curves::twisted_edwards::AffineVar as TEAffineVar;
-use ark_r1cs_std::ToConstraintFieldGadget;
+use ark_r1cs_std::{ToBytesGadget, ToConstraintFieldGadget};
 use ark_relations::r1cs::SynthesisError;
 use ark_std::vec;
 use ark_std::vec::Vec;
-/// An interface for objects that can be absorbed by a `CryptographicSpongeVar`.
+/// An interface for objects that can be absorbed by a `CryptographicSpongeVar` whose constraint field
+/// is `CF`.
 pub trait AbsorbGadget<F: PrimeField> {
+    /// Converts the object into a list of bytes that can be absorbed by a `CryptographicSpongeVar`.
+    /// return the list.
+    fn to_sponge_bytes_var(&self) -> Result<Vec<UInt8<F>>, SynthesisError>;
```

Comment:
I would remove the `_var`; these methods can only be called on variables anyway, so the extra reminder is kind of redundant.

Reply:
fixed